grafana-plugin/src/containers/RotationForm/parts/DateTimePicker.tsx

```diff
@@ -27,40 +26,56 @@ interface DateTimePickerProps {
 export const DateTimePicker = observer(
   ({ value: propValue, onChange, disabled, onFocus, onBlur, error }: DateTimePickerProps) => {
     const styles = useStyles2(getStyles);
-    const {
-      timezoneStore: { getDateInSelectedTimezone },
-    } = useStore();
-    const valueInSelectedTimezone = getDateInSelectedTimezone(propValue);
-    const valueAsDate = valueInSelectedTimezone.toDate();
-
-    const handleDateChange = (newDate: Date) => {
-      const localMoment = getDateInSelectedTimezone(dayjs(newDate));
-      const newValue = localMoment
-        .set('year', newDate.getFullYear())
-        .set('month', newDate.getMonth())
-        .set('date', newDate.getDate())
-        .set('hour', valueAsDate.getHours())
-        .set('minute', valueAsDate.getMinutes())
-        .set('second', valueAsDate.getSeconds());
-
-      onChange(newValue);
+    const { timezoneStore } = useStore();
+    const { selectedTimezoneOffset } = timezoneStore;
+
+    const handleDateChange = (value: Date) => {
+      const newDate = dayjs(value)
+        .utcOffset(selectedTimezoneOffset)
```

Comment:
In all the places when we do `someDayjsDate.utcOffset(timezoneStore.selectedTimezoneOffset)`, can we use `timezoneStore.getDateInSelectedTimezone(someDayjsDate)` instead? It works the same, right? We do it already in other places so we could be consistent

Reply:
I slightly improved this part since the last time you checked. I added a helper `toDateWithTimezoneOffset`

```jsx
export const toDateWithTimezoneOffset = (date: dayjs.Dayjs, timezoneOffset: number) => {
  return date.utcOffset() === timezoneOffset ? date : date.utcOffset(timezoneOffset);
};
```

The idea is that if you ever apply the offset twice on a dayjs object, it might have some weird implications like getting weird format or yielding weird results for `hour()` and `minute()`. So instead of appending `utcOffset` to those calls, I rely on this method that checks for an already applied offset